source/features/last-notification-page-button.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager';
+import looseParseInt from '../helpers/loose-parse-int';
+
+function init(): void | false {
+	const nextButton = select('a[data-hotkey="ArrowRight"]');
+	if (!nextButton) {
+		return false;
+	}
+
+	const lastNotificationPageNode = select('.js-notifications-list-paginator-counts')!.lastChild!;
+	const lastNotificationPageNumber = looseParseInt(lastNotificationPageNode);
+	const lastCursor = Math.floor(lastNotificationPageNumber / 50) * 50;
+	const nextButtonSearch = new URLSearchParams(nextButton.search);
+	nextButtonSearch.set('after', btoa('cursor:' + String(lastCursor)));
+	nextButton.search = nextButtonSearch.toString();
```

Comment:
This changes the next button too. Please check my suggestions carefully, I wrote some [specific code](https://github.com/refined-github/refined-github/pull/6133#discussion_r1013518601) that isn't this. Let's not do 40-comments reviews again :(

```suggestion
```

Reply:
https://github.com/refined-github/refined-github/pull/6133#discussion_r1013878505

Reply:
I fail to see how that would not work. `String(nextButtonSearch)` is exactly the same as `nextButtonSearch.toString()` and the search is exactly what we need.

Reply:
Now I understand what you are saying. 👍

Reply:
Your original suggestion was to remove the nextButtonURL, which I did.

If I don't include this line then the URL will only be the search, `https://github.com/after=Y3Vyc29yOjEwNTA%3D&query=is%3Adone` missing the notifications URL.

I feel bad we are going though so many comments.

If you feel that I am missing something, I have no issue with you closing and allowing someone else to pick this up. (No hard feelings I promise)

Reply:
Gotcha, the only misunderstanding here is that `?` is missing when using `String()` whereas it exists in `url.search`. We just need to prepend the question mark so that the attribute will be exactly `"?after=Y3Vyc29yOjEwNTA%3D&query=is%3Adone"`, which is relative to the current pathname.

Possible ways, depending on which one TS likes the most

```js
 			href={'?' + String(nextButtonSearch)}
```
```js
 			href={`?${nextButtonSearch}`}
```